include/aspect/material_model/interface.h

```diff
@@ -928,6 +928,42 @@ namespace aspect
     compute_volume_fractions(const std::vector<double> &compositional_fields,
                              const ComponentMask &field_mask = ComponentMask());
 
+
+    /**
+     * For multicomponent material models:
+     * Enumeration for selecting which averaging scheme to use when
+     * averaging the properties of different compositional fields.
+     * Select between harmonic, arithmetic, geometric, and
+     * maximum_composition. The max composition scheme simply uses the
+     * viscosity of whichever field has the highest volume fraction.
+     */
+    enum CompositionalAveragingOperation
+    {
+      harmonic,
+      arithmetic,
+      geometric,
+      maximum_composition
+    };
+
+    /**
+     * For multicomponent material models:
+     * Material models compute output quantities such as the viscosity, the
+     * density, etc. For some models, these values depend strongly on the
+     * composition, and more than one compositional field might have nonzero
+     * values at a given quadrature point. This means that properties have to
+     * be averaged based on the fractions of each compositional field present.
+     * This function performs this type of averaging, and allows it to choose
+     * between harmonic, arithmetic and geometric averaging or simply selecting
+     * the material property of the compositional field with the largest volume
+     * fraction. For each quadrature point, averaging is conducted over the
+     * compositional fields given in @p volume_fractions, plus an additional
```

Comment:
quadrature point doesn't make sense here.
Does it make sense to have this function do the work over n points at the same time?
Also, can you state the expected lengths of the 2 vectors?

Reply:
At the moment, this function is used inside the loop over all points the evaluate function gets, and in between other operations that compute things for just on point. So, for example, the parameter values have to be computed first, that's done inside of the loop over all points too, and then this function is called to average them (and some material models have several calls to this function to average different material properties). So I think this function is more useful if it just operates on one point.